ospf6d/ospf6_area.c

```diff
@@ -454,6 +455,31 @@ void ospf6_area_show(struct vty *vty, struct ospf6_area *oa,
 		json_object_int_add(json_area, "numberOfAreaScopedLsa",
 				    oa->lsdb->count);
 
+		json_lsa_stats = json_object_new_object();
+		json_object_int_add(json_lsa_stats, "numberOfRouterLsa",
+				    oa->lsdb->stats.num_type1);
+		json_object_int_add(json_lsa_stats, "numberOfNetworkLsa",
+				    oa->lsdb->stats.num_type2);
+		json_object_int_add(json_lsa_stats,
+				    "numberOfInterAreaPrefixLsa",
+				    oa->lsdb->stats.num_type3);
+		json_object_int_add(json_lsa_stats,
+				    "mumberOfInterAreaRouterLsa",
+				    oa->lsdb->stats.num_type4);
+		json_object_int_add(json_lsa_stats, "numberOfAsExternalLsa",
+				    oa->lsdb->stats.num_type5);
+		json_object_int_add(json_lsa_stats, "numberOfNssaLsa",
+				    oa->lsdb->stats.num_type7);
+		json_object_int_add(json_lsa_stats, "numberOfLinkLsa",
+				    oa->lsdb->stats.num_type8);
+		json_object_int_add(json_lsa_stats,
+				    "numberOfIntraAreaPrefixLsa",
+				    oa->lsdb->stats.num_type9);
+		json_object_int_add(json_lsa_stats, "numberOfUnknownLsa",
+				    oa->lsdb->stats.num_unknown);
+		json_object_object_add(json_area, "lsaStatistics",
+				       json_lsa_stats);
+
```

Comment:
replace the above with:
```
json_lsa_stats = JSON_OBJECT_NEW_ARRAY(json_object_new_int, oa->lsdb->lsaTypeCounts, sizeof(os->lsdb->lsaTypeCounts));
json_object_object_add(json_area, "lsaStatistics", json_lsa_stats);
```
and put something like this in lib/json.h
```
#define JSON_OBJECT_NEW_ARRAY(jsonf, carray, len)                              \
	({                                                                     \
		struct json_object *_o = json_object_new_array();              \
		for (int _i = 0; _i < (len); _i++)                             \
			json_object_array_add(_o, (jsonf)((carray)[_i]));      \
		(_o);                                                          \
	})
```

Reply:
I would like to keep the output in key/value pairs. Otherwise i would have to add the same values in the text output to make them somewhat useful to human users.

Reply:
So this is what one might call "fragile code" it's code that will be wrong if another type is added but the developer doesn't notice they have to update this show command. By using a loop and the #define contants that never happens.

I'm thinking any human who is looking by hand at the number of a specific type of LSA isn't going to have trouble indexing into the array for the type they are looking for. 

If you really want to keep the key/value then I'd push for the key being the number "1", and still using a loop to do this code.

Reply:
How many additional LSA types do you expect to be specified over the next 5 years?

Reply:
Is this a serious question? If so what's special about 5 years?

Reply:
No. / Nothing.

I like two things:
1. Seeing the world burn.
2. Descriptive variable names.

Are you just going to keep insisting I change the cosmetics of the JSON output or can we move on with this PR? I implemented all other changes you suggested, but have no trouble closing it if you are not willing to reach a compromise.

Reply:
Don't want to be a part of the conflict, posting just to let everyone know about proposed/current standards.

If you look at the draft for OSPF YANG model (https://datatracker.ietf.org/doc/html/draft-ietf-ospf-yang-29), you'll see that LSA statistics are implemented as a list. And a YANG list should be represented as a JSON array according to RFC 7951 (https://datatracker.ietf.org/doc/html/rfc7951#section-5.4).

Reply:
> No. / Nothing.
> 
> I like two things:
> 
> 1. Seeing the world burn.
> 2. Descriptive variable names.
> 
> Are you just going to keep insisting I change the cosmetics of the JSON output or can we move on with this PR? I implemented all other changes you suggested, but have no trouble closing it if you are not willing to reach a compromise.

tl;dr I certainly don't want you to "take your ball and go home" (as the saying goes) over this, so commit whatever you think is the best/workable version I guess. I'd at least request that you give read to the below first though.

As with my other suggestions this is ultimately based on the fact that JSON is for machine parsing. If I'm writing a client and the numerical type value is the key, I don't need a translation table for FRR specific strings to try and find the LSA type I want -- the value is standardized. For show output targeted for human consumption using a descriptive string makes sense, but for JSON we're now forcing the client to write a translator to go from numerical type value to the strings you picked. Also, to support more LSA types both the client and the server would need to be updated in sync with each other.

To answer your earlier question, while I don't know when support may be added to FRR, there are currently 14 other LSA types defined that aren't represented here. https://www.iana.org/assignments/ospfv3-parameters/ospfv3-parameters.xhtml#ospfv3-parameters-3

Reply:
@davischw you marked it as resolved, does it mean that you won't reconsider the implementation?

Reply:
@idryzhov Yes.